### PR TITLE
Pin flow

### PR DESCRIPTION
**Pull request by n0rdy**

### CODE REVIEW

File: src/main/java/de/adorsys/xs2a/adapter/ui/service/builder/RequestBuilder.java [hunk 1]
```diff
@@ -59,4 +64,31 @@ public ObjectNode startAuthorisationBody() {
 
         return headers;
     }
+
+    public ObjectNode startAuthorisationWithPsuAuthenticationBody(String pin, boolean encrypted) {
+        PsuData psuData = new PsuData();
+
+        if (encrypted) {
+            psuData.setEncryptedPassword(pin);
```
Comment (spetrychenko) [inline src/main/java/de/adorsys/xs2a/adapter/ui/service/builder/RequestBuilder.java]: should we encrypt the password by ourselves in this case?

Comment (n0rdy) [reply src/main/java/de/adorsys/xs2a/adapter/ui/service/builder/RequestBuilder.java]: I thought about that. 

- For sandbox we shouldn't (at least for Deutsche Bank)
- For real banks - have no idea, as we haven't faced such behavior yet.

Anyway, there is another question: what exact algorithm should be used for the encryption then?

Comment (spetrychenko) [reply src/main/java/de/adorsys/xs2a/adapter/ui/service/builder/RequestBuilder.java]: and what is BG Spec says about that?

Comment (n0rdy) [reply src/main/java/de/adorsys/xs2a/adapter/ui/service/builder/RequestBuilder.java]: Sounds like it depends from the ASPSP implementation
image

Comment (spetrychenko) [reply src/main/java/de/adorsys/xs2a/adapter/ui/service/builder/RequestBuilder.java]: i see

**Pull request merged**